analyzers/src/SonarAnalyzer.CSharp/Helpers/CountComparisonSyntaxHelper.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace SonarAnalyzer.Helpers
+{
+    public static class CountComparisonSyntaxHelper
+    {
+        private static readonly IExpressionNumericConverter Converter = new CSharpExpressionNumericConverter();
+
+        public static CountComparisonSyntax CountComparison(this BinaryExpressionSyntax binary, SemanticModel semanticModel)
+        {
+            int? left = default;
+            int? right = default;
+            if (Converter.TryGetConstantIntValue(binary.Left, semanticModel, out var l_out))
+            {
+                left = l_out;
+            }
```

Comment:
It took me a while to actually understand this. Let's make it more direct and explicit
```suggestion
            int? left = Converter.TryGetConstantIntValue(binary.Left, semanticModel, out var leftValue) ? leftValue : null;
```